.github/workflows/comment_bot.yml

```diff
@@ -66,8 +66,9 @@ jobs:
           DEFAULT_BRANCH=${{ github.event.repository.default_branch }}
           git remote add upstream https://github.com/apache/arrow
           git fetch upstream
+
           changed() {
-            git diff --name-only HEAD..upstream/$DEFAULT_BRANCH | grep -e "$1" >/dev/null 2>&1
+            git diff --name-only upstream/$DEFAULT_BRANCH... | grep -e "$1" >/dev/null 2>&1
```

Comment:
LGTM, I've tested myself on a local branch and a master branch with new commits. Old command:
```
$ git diff --name-only HEAD..upstream/master
c_glib/Brewfile
dev/tasks/homebrew-formulae/apache-arrow-glib.rb
r/R/dplyr-summarize.R
r/src/type_infer.cpp
r/tests/testthat/test-dplyr-summarize.R
r/tests/testthat/test-type.R
```
new command output expected only changes on the PR (not on master):
```
$ git diff --name-only upstream/master...
c_glib/Brewfile
dev/tasks/homebrew-formulae/apache-arrow-glib.rb

```